kernel/driver.cc

```diff
@@ -80,20 +80,37 @@ USING_YOSYS_NAMESPACE
 #ifdef EMSCRIPTEN
 #  include <sys/stat.h>
 #  include <sys/types.h>
+#  include <emscripten.h>
 
-extern "C" int main(int, char**);
+extern "C" int main(int, const char**);
 extern "C" void run(const char*);
 extern "C" const char *errmsg();
 extern "C" const char *prompt();
 
-int main(int, char**)
+int main(int argc, const char**argv)
```

Comment:
That's an odd. Why is the `const` needed? According to 5.1.2.2.1 of ISO C99, `main()` must be declared as `int main(void)` or `int main(int argc, char *argv[])` or equivalent. (GNU libc adds `int main (int argc, char *argv[], char *envp[])`.)